docs(target): Introduce direct address targets and explain tradeoffs

hugoamvieira opened the pull request.

johanbrandhorst commented on website/content/docs/concepts/domain-model/index.mdx:
```diff
@@ -52,7 +52,26 @@ The diagram below illustrates the relationship between the different IAM compone
+address or DNS name directly on the target resource. This mechanism provides
+benefits for new Boundary users, as it's less complex to initially set-up and
+understand, or users who don't require Boundary's flexibility and scalability.
+Because of its simplicity, using this path on environemnts where a significant
```
```suggestion
Because of its simplicity, using this path on environments where a significant
```

hugoamvieira replied:
Haha thanks for spotting, fixed with [c628d10](https://github.com/hashicorp/boundary/pull/2708/commits/c628d10cd187f8afaa470911c764051a56b7f4fa)

Dan-Heath commented on website/content/docs/concepts/domain-model/index.mdx:
```diff
@@ -52,7 +52,26 @@ The diagram below illustrates the relationship between the different IAM compone
+- Target with an address
+- Target with host sources
+
+A target with an address represents a target that is set-up with a single IP
```
```suggestion
A target with an address represents a target that is set up with a single IP
```

Dan-Heath commented on website/content/docs/concepts/domain-model/index.mdx:
```diff
@@ -52,7 +52,26 @@ The diagram below illustrates the relationship between the different IAM compone
+- Target with host sources
+
+A target with an address represents a target that is set-up with a single IP
+address or DNS name directly on the target resource. This mechanism provides
+benefits for new Boundary users, as it's less complex to initially set-up and
+understand, or users who don't require Boundary's flexibility and scalability.
```
```suggestion
address or DNS name directly on the target resource. This mechanism provides
benefits for new Boundary users and users who don't require Boundary's flexibility and scalability, as it's less complex to set up and understand.
```

Dan-Heath commented on website/content/docs/concepts/domain-model/index.mdx:
```diff
@@ -52,7 +52,26 @@ The diagram below illustrates the relationship between the different IAM compone
+A target with host sources represents a target that has one or more [host
+sets][] associated with it. It trades off the simplicity of directly setting an
+address on a target with Boundary's flexibility and scalability for discovering
+and categorizing networked resources. This is the recommended path for setting
+up Boundary on environments where a significant number of resources exist.
```
```suggestion
and categorizing networked resources. We recommend this path for setting
up Boundary in environments where a significant number of resources exist.
```

Dan-Heath commented on website/content/docs/oss/installing/connect-to-dev-target.mdx:
```diff
@@ -8,16 +8,32 @@ description: |-
-connect` is executed against this target, the single available host will be
-selected and a local authenticated proxy to the host is established on the
+corresponds to the default SSH port using TCP). This target is set up with
+`127.0.0.1` as its address. When `boundary connect` is executed against this
```
```suggestion
`127.0.0.1` as its address. When you execute `boundary connect` against this
```

Dan-Heath commented on website/content/docs/oss/installing/connect-to-dev-target.mdx:
```diff
@@ -8,16 +8,32 @@ description: |-
 
+There is also a secondary `tcp` target with (`ttcp_0987654321`) which uses host
+sets instead of an address. The host sets for this target contain the default
+host, which has the address 127.0.0.1. When boundary connect is executed against
```
```suggestion
host, which has the address 127.0.0.1. When you execute `boundary connect` against
```

Dan-Heath commented on website/content/docs/oss/installing/connect-to-dev-target.mdx:
````diff
@@ -8,16 +8,32 @@ description: |-
+```
+
+For the purpose of this guide, both of these targets accomplish the same goal,
+however they have different purposes. Please see the our
````
```suggestion
however they have different purposes. Refer to the 
```

Dan-Heath reviewed (commented):
I added a few suggestions for style and flow, but I think this makes sense and reads well. Please let me know if you have any questions. Thank you!

hugoamvieira commented:
Thanks a lot @Dan-Heath, I think I've addressed all your suggestions with [c628d10](https://github.com/hashicorp/boundary/pull/2708/commits/c628d10cd187f8afaa470911c764051a56b7f4fa). Let me know if I missed something or you have more suggestions :)

Dan-Heath reviewed (approved):
LGTM, thank you @hugoamvieira !
